src/gsy_e/models/strategy/heat_pump.py

```diff
@@ -113,37 +117,38 @@ def post_order(
 
         if order_energy_kWh <= FLOATING_POINT_TOLERANCE:
             return
+
         market.bid(
             order_rate * order_energy_kWh, order_energy_kWh,
-            buyer=self.owner.name,
             original_price=order_rate * order_energy_kWh,
-            buyer_origin=self.owner.name,
-            buyer_origin_id=self.owner.uuid,
-            buyer_id=self.owner.uuid,
+            buyer=TraderDetails(self.owner.name, self.owner.uuid,
+                                self.owner.name, self.owner.uuid),
             time_slot=market_slot)
 
     def remove_open_orders(self, market: "MarketBase", market_slot: DateTime):
         if self.area.is_market_spot(market.id):
             bids = [bid
                     for bid in market.bids.values()
-                    if bid.buyer == self.owner.name]
+                    if bid.buyer.name == self.owner.name]
         else:
             bids = [bid
                     for bid in market.slot_bid_mapping[market_slot]
-                    if bid.buyer == self.owner.name]
+                    if bid.buyer.name == self.owner.name]
 
         for bid in bids:
             market.delete_bid(bid)
 
     def _get_energy_buy_energy(self, buy_rate: float, market_slot: DateTime) -> float:
         if buy_rate > self.preferred_buying_rate:
-            return self._energy_params.energy_loss_kWh.get(market_slot, 0)
-        return self._energy_params.energy_demand_kWh.get(market_slot, 0)
+            return self._energy_params.energy_loss_kWh.get(market_slot, 1)
```

Comment:
@mrsaemir I think this comment is not valid any more right?
The energy parameters of the `HeatPump EnergyParamters` deal with populating the `min_energy_demand_kWh` and `max_energy_demand_kWh`. They can only be 0 if the heatpump does not consume energy. But this case is caught in the `post_order` method. 
Having both values in one profile does not work, because the price changes over the frame of the time_slot. As soon as the `preferred_buying_rate` is reached, the other energy value need to be accessed and not calculated on the fly. 
I hope this makes sense.